src/basic/fileio.c

```diff
@@ -1429,6 +1429,7 @@ int rename_and_apply_smack_floor_label(const char *from, const char *to) {
                 return -errno;
 
 #if HAVE_SMACK_RUN_LABEL
+#include "smack-util.h"
```

Comment:
please move this to the top of the file

Reply:
I did but LGTM was complaining `smack-util.h` not found - false positive?

Reply:
I guess this is caused by `smack-util.h` being in `src/shared/`, and files from there shouldn't be included from stuff under `src/basic/` (see https://github.com/systemd/systemd/blob/main/docs/ARCHITECTURE.md#shared-code).

Reply:
right, problem with the new `fileio.c` is that it depends on `SMACK_FLOOR_LABEL` et al. which are defined in `smack-util.h`

Reply:
yes indeed, I guess this whole function should move to smack-util.c (outside of the ifdef)

Reply:
Problem is that `rename_and_apply_smack_floor_label` is a two-step operation - it renames a file and then apply the SMACK floor label.

Should we just separate it into 2 functions and fix all other invocations of `rename_and_apply_smack_floor_label`?

Reply:
I think it's fine to move the whole function, it's used only from tools